packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -606,7 +606,7 @@ export const useArbTokenBridge = (
   }) {
     const depositRequest = await erc20Bridger.getDepositRequest({
       // Setting `amount` to zero so it doesn't fail on not enough allowance
-      amount: BigNumber.from(0),
+      amount: amount ?? BigNumber.from(0),
```

Comment:
Not sure if I understand this approach completely.
`Setting `amount` to zero so it doesn't fail on not enough allowance`

Why was this explicitly hard-coded in the first place if it wasn't full-proof?

Reply:
My guess is because previously we allowed undefined for amount from the query parameter? Not entirely sure. It was from #457

Reply:
I'm not exactly sure if that's the case but the actual amount doesn't matter for gas estimation on nitro, that's why we set it to 0 so that it doesn't run into an issue where there's not enough allowance set. I believe this should be restructured in the sdk itself if that's the case.

I wouldn't make any changes here without making sure with @spsjvc or @yahgwai